### PR TITLE
Add Metadata Converter

### PR DESCRIPTION
**Pull request by justkawal**
## Any follow ups?
Please review PR: #121, before reviewing this PR.

## Description
* Adds Metadata-V14 to Chain-Description Conversion
* Adds Metadata-Legacy  to Chain-Description Conversion
* Adds Cached Library to cache the processed outputs for faster computations.

## Testing Plan 
`dart test`

### CODE REVIEW

File: packages/substrate_metadata/lib/chain_description/parse_legacy.dart [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    _registry = scale_codec.TypeRegistry(
+        types: legacyTypes.types, typesAlias: legacyTypes.typesAlias);
+    _defineGenericExtrinsicEra();
+    _defineGenericLookupSource();
+    _defineOriginCaller();
+    _defineGenericCall();
+    _defineGenericEvent();
+    _defineGenericSignature();
+   }
+   */
```
Comment (Lohann) [inline packages/substrate_metadata/lib/chain_description/parse_legacy.dart:32]: Create an issue for this and remove this commented code.
```suggestion
```

File: packages/substrate_metadata/lib/chain_description/parse_v14.dart [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                    path: info['path'],
+                    docs: info['docs'],
+                  );
+                case 'Compact':
+                  // ignore: unnecessary_cast
+                  return scale_codec.CompactType(
+                    type: (def as Si1TypeDef_Compact).value.type,
+                    path: info['path'],
+                    docs: info['docs'],
+                  ) as scale_codec.Type;
```
Comment (Lohann) [inline packages/substrate_metadata/lib/chain_description/parse_v14.dart:247]: It says the cast is unnecessary, so why not simply remove the type cast?

**Review by Lohann — CHANGES_REQUESTED** [review]

**Comment by pastre** [comment]
When messing with codegen, please open a pull request with the generated code only and describe which command you ran to generate the code. Open other pull requests with your modifications later

File: packages/substrate_metadata/lib/chain_description/parse_legacy.dart (hunk 1 above)
Comment (pastre) [reply packages/substrate_metadata/lib/chain_description/parse_legacy.dart:32]: I'm fine with removing the comment when the issue is resolved

File: packages/substrate_metadata/lib/chain_description/parse_v14.dart (hunk 2 above)
Comment (justkawal) [reply packages/substrate_metadata/lib/chain_description/parse_v14.dart:247]: Its an issue in VS-Code editor, I don't if this issue is coming in everyone's PC or not.
It throw me error if I do not type-cast.

**Review by Lohann — APPROVED** [review]

**Comment by justkawal** [comment]
@leonardocustodio 
Can you please review it ?

**Review by leonardocustodio — APPROVED** [review]

**Pull request merged**